### ISSUE TITLE
How to plot efficient frontier with constraints

### ISSUE THREAD

**Issue by hnicks13** [robertmartin8/PyPortfolioOpt#135]
I understand that the EfficientFrontier class does not support plotting and I have to use CLA. However, my understanding is that with CLA you cannot add extra constraints and also it does not support the efficient_risk() method, i.e. it does not return the optimal portfolio for a given target risk (it only supports minimum vol and max sharpe). So how do I go about it if I want to plot the whole frontier AND return optimal portfolios for various risk levels AND add extra constraints ? Any help is appreciated.

**Comment by robertmartin8** [comment]
@hnicks13 

If you want to return optimal portfolios and visualise the constrained efficient frontier, you will have to generate the plot manually. This is an example of how to do it:

```python
rets = []
stds = []
for risk_aversion in np.arange(0, 2, 0.05):
    ef = EfficientFrontier(mu, S)
    ef.add_constraints(...) 
    ef.max_quadratic_utility(risk_aversion)
    ret, sigma, _ = ef.portfolio_performance()
    rets.append(ret)
    std.append(sigma)

plt.plot(stds, rets)
```

I am currently trying to think of a way to implement this in the `plotting` module. Hope this helps!

Robert

**Comment by hnicks13** [comment]
@robertmartin8 Great, I'll give that a go! Many thanks for your help!

**Comment by hnicks13** [comment]
Just to confirm that this works fine. Thanks for the code!

**Comment by robertmartin8** [comment]
Cheers for letting me know!

I'll keep the issue up because it's something that I want to provide by
default in future.

**Comment by robertmartin8** [comment]
This has been implemented in v1.3.1 🚀 

See [docs](https://pyportfolioopt.readthedocs.io/en/latest/Plotting.html) for an example :)
